src/webgpu/api/validation/queue/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,480 @@
+- For source.imageBitmap:
+  - imageBitmap generated from ImageData:
+    - Check that an error is generated when imageBitmap is closed. Otherwise, no error should be
+      generated.
+
+- For destination.texture:
+  - For 2d destination textures:
+    - Check that an error is generated when texture is in destroyed state. Otherwise, no error should
+      be generated.
+    - Check that an error is generated when texture is an error texture. Otherwise, no error should
+      be generated.
+    - Check that an error is generated when texture is created without usage COPY_DST. Otherwise,
+      no error should be generated.
+    - Check that an error is generated when sample count is not 1. Otherwise, no error should be
+      generated.
+    - Check that an error is generated when mipLevel is too large. Otherwise, no error should be
+      generated.
+    - Check that an error is generated when texture format is not valid. Otherwise, no error should
+      be generated.
+
+- For copySize:
+  - Noop copy shouldn't throw any exception or return any validation error.
+  - Check that an error is generated when destination.texture.origin + copySize is too large.
+    Otherwise, no error should be generated.
+
+TODO: 1d, 3d texture and 2d array textures.
+`;
+
+import { poptions, params } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { kAllTextureFormats, kTextureUsages } from '../../../capability_info.js';
+import { ValidationTest } from '../validation_test.js';
+
+const kDefaultBytesPerPixel = 4; // using 'bgra8unorm' or 'rgba8unorm'
+const kDefaultWidth = 32;
+const kDefaultHeight = 32;
+const kDefaultDepth = 1;
+const kDefaultMipLevelCount = 6;
+
+// From spec
+const kValidTextureFormatsForCopyIB2T = [
+  'rgba8unorm',
+  'rgba8unorm-srgb',
+  'bgra8unorm',
+  'bgra8unorm-srgb',
+  'rgb10a2unorm',
+  'rgba16float',
+  'rgba32float',
+  'rg8unorm',
+  'rg16float',
+];
+
+function getMipMapSize(width: number, height: number, mipLevel: number) {
+  return {
+    mipWidth: Math.max(width >> mipLevel, 1),
+    mipHeight: Math.max(height >> mipLevel, 1),
+  };
+}
+
+interface WithMipLevel {
+  mipLevel: number;
+}
+
+interface WithMipLevelSizeAndOrigin extends WithMipLevel {
+  srcImageSize: GPUExtent3DDict;
+  dstTextureSize: GPUExtent3DDict;
+  srcOriginValue: GPUOrigin2DDict;
+  dstOriginValue: GPUOrigin3DDict;
+}
+
+// Helper function to generate dst origin value to take mipLevel into consideration.
+function generateDstOriginValue({ mipLevel }: WithMipLevel) {
+  const origin = getMipMapSize(kDefaultWidth, kDefaultHeight, mipLevel);
+
+  return poptions('dstOriginValue', [
+    { x: 0, y: 0, z: 0 },
+    { x: origin.mipWidth, y: 0, z: 0 },
+    { x: 0, y: origin.mipHeight, z: 0 },
+    { x: 0, y: 0, z: kDefaultDepth },
+  ]);
+}
+
+// Helper function to generate copySize to take mipLevel, origin into consideration
+function generateCopySize({
```

Comment:
This could generate a copySizes just for a single target texture level, with the rework of the OOB tests suggested below.

Reply:
Left this one for double checking whether it is resolved.